devimint/src/devfed.rs

```diff
@@ -305,4 +347,45 @@ impl DevJitFed {
             electrs: self.electrs().await?.to_owned(),
         })
     }
+
+    pub async fn fast_terminate(self) {
+        let Self {
+            bitcoind,
+            cln,
+            lnd,
+            fed,
+            gw_cln,
+            gw_lnd,
+            electrs,
+            esplora,
+            ..
+        } = self;
+
+        join!(
+            async move {
+                drop(gw_cln);
+            },
+            async move {
+                drop(gw_lnd);
+            },
+            async move {
+                drop(fed);
+            },
+            async move {
+                drop(lnd);
+            },
+            async move {
+                drop(cln);
+            },
+            async move {
+                drop(esplora);
+            },
+            async move {
+                drop(electrs);
+            },
+            async move {
+                drop(bitcoind);
+            },
+        );
```

Comment:
how is this even faster? drop functions need to be async for this to work concurrently

Reply:
They are async due to `block_in_place` trick, right?

Reply:
still in the end, only single blocking code function can run on the thread in the end?

and all drop functions are blocking

Reply:
try replacing a bunch of them with

```rust
           async move {
                info!("drop start: gw_cln");
                drop(gw_cln);
                info!("drop end: gw_cln");
           },
```

I would be really surprised if start and end get intermixed

async blocks don't yield without .await :shrug:

Reply:
@maan2003 Why do you have to be so right? :P

Reply:
I moved the drop to own async task and I think it works now. I'm testing it with fedimintd paralle one dropped, to make it slow and I see:

```
2024-05-13T02:43:08.300493Z  INFO devimint::devfed: drop start
2024-05-13T02:43:08.300528Z  INFO devimint::devfed: drop start
2024-05-13T02:43:08.300535Z  INFO devimint::devfed: drop start
2024-05-13T02:43:08.300544Z  INFO devimint::devfed: drop start
2024-05-13T02:43:08.300553Z  INFO devimint::devfed: drop start
2024-05-13T02:43:08.300561Z  INFO devimint::devfed: drop start
2024-05-13T02:43:08.300568Z  INFO devimint::devfed: drop start
2024-05-13T02:43:08.300576Z  INFO devimint::devfed: drop start
2024-05-13T02:43:08.300609Z DEBUG spawn{task="spawn_drop"}: fm::devimint: sending signal to terminate child process name=gatewayd-cln signal="SIGTERM"
2024-05-13T02:43:08.300618Z  INFO devimint::devfed: drop end
2024-05-13T02:43:08.300631Z DEBUG spawn{task="spawn_drop"}: fm::devimint: sending signal to terminate child process name=fedimintd-0 signal="SIGTERM"
2024-05-13T02:43:08.300651Z  INFO devimint::devfed: drop end
2024-05-13T02:43:08.300663Z  INFO devimint::devfed: drop end
2024-05-13T02:43:08.300683Z DEBUG spawn{task="spawn_drop"}: fm::devimint: sending signal to terminate child process name=gatewayd-lnd signal="SIGTERM"
2024-05-13T02:43:08.300718Z DEBUG spawn{task="spawn_drop"}: fm::devimint: sending signal to terminate child process name=esplora signal="SIGTERM"
2024-05-13T02:43:08.300772Z DEBUG spawn{task="spawn_drop"}: fm::devimint: sending signal to terminate child process name=electrs signal="SIGTERM"
2024-05-13T02:43:08.303644Z  INFO devimint::devfed: drop end
2024-05-13T02:43:08.306211Z  INFO devimint::devfed: drop end
2024-05-13T02:43:09.090963Z DEBUG spawn{task="spawn_drop"}: fm::devimint: sending signal to terminate child process name=fedimintd-1 signal="SIGTERM"
2024-05-13T02:43:09.913950Z DEBUG spawn{task="spawn_drop"}: fm::devimint: sending signal to terminate child process name=fedimintd-2 signal="SIGTERM"
2024-05-13T02:43:10.301991Z DEBUG spawn{task="spawn_drop"}: fm::devimint: sending signal to terminate child process name=gatewayd-cln signal="SIGKILL"
2024-05-13T02:43:10.302009Z DEBUG spawn{task="spawn_drop"}: fm::devimint: sending signal to terminate child process name=gatewayd-lnd signal="SIGKILL"
2024-05-13T02:43:10.310133Z DEBUG spawn{task="spawn_drop"}: fm::devimint: > lightning-cli --network regtest --lightning-dir=/tmp/devimint-4113675-621/cln plugin stop gateway-cln-extension
2024-05-13T02:43:10.310798Z DEBUG spawn{task="spawn_drop"}: fm::devimint: sending signal to terminate child process name=lnd signal="SIGTERM"
2024-05-13T02:43:10.319800Z DEBUG spawn{task="spawn_drop"}: fm::devimint: sending signal to terminate child process name=lightningd signal="SIGTERM"
2024-05-13T02:43:10.320456Z  INFO devimint::devfed: drop end
2024-05-13T02:43:10.420133Z  INFO devimint::devfed: drop end
2024-05-13T02:43:10.751749Z DEBUG spawn{task="spawn_drop"}: fm::devimint: sending signal to terminate child process name=fedimintd-3 signal="SIGTERM"
2024-05-13T02:43:11.580099Z DEBUG spawn{task="spawn_drop"}: fm::devimint: sending signal to terminate child process name=bitcoind signal="SIGTERM"
2024-05-13T02:43:11.613527Z  INFO devimint::devfed: drop end
2024-05-13T02:43:11.613581Z DEBUG fm::devimint: Finished elapsed_ms=85858
```